src/compiler/checker.ts

```diff
@@ -15201,22 +15201,22 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return true;
     }
 
-    function getSignatureOfTypeTag(node: SignatureDeclaration | JSDocSignature) {
+    function getSignatureOfTypeTag(node: SignatureDeclaration | JSDocSignature, allowMembers: boolean) {
         // should be attached to a function declaration or expression
         if (!(isInJSFile(node) && isFunctionLikeDeclaration(node))) return undefined;
         const typeTag = getJSDocTypeTag(node);
-        return typeTag?.typeExpression && getSingleCallSignature(getTypeFromTypeNode(typeTag.typeExpression));
+        return typeTag?.typeExpression && getSingleCallSignature(getTypeFromTypeNode(typeTag.typeExpression), allowMembers);
     }
 
-    function getParameterTypeOfTypeTag(func: FunctionLikeDeclaration, parameter: ParameterDeclaration) {
-        const signature = getSignatureOfTypeTag(func);
+    function getParameterTypeOfTypeTag(func: FunctionLikeDeclaration, parameter: ParameterDeclaration, allowMembers: boolean) {
+        const signature = getSignatureOfTypeTag(func, allowMembers);
         if (!signature) return undefined;
         const pos = func.parameters.indexOf(parameter);
         return parameter.dotDotDotToken ? getRestTypeAtPosition(signature, pos) : getTypeAtPosition(signature, pos);
     }
 
     function getReturnTypeOfTypeTag(node: SignatureDeclaration | JSDocSignature) {
-        const signature = getSignatureOfTypeTag(node);
+        const signature = getSignatureOfTypeTag(node, /*allowMembers*/ false);

```

Comment:
The only function that calls `getReturnTypeOfTypeTag` is `getReturnTypeFromAnnotation`. That one uses `getEffectiveReturnTypeNode` and roughly returns that as is if it's available. An explicit annotated return type might appear in a function with members - so members are kinda allowed in that code path. So I think they should be allowed here too.